advisor/src/main/kotlin/advisors/NexusIq.kt

```diff
@@ -83,7 +83,7 @@ class NexusIq(name: String, private val nexusIqConfig: NexusIqConfiguration) : A
     override suspend fun retrievePackageFindings(packages: Set<Package>): Map<Package, List<AdvisorResult>> {
         val startTime = Instant.now()
 
-        val components = packages.map { pkg ->
+        val components = packages.filter { it.purl.isNotBlank() }.map { pkg ->
```

Comment:
Would `isNotEmpty()` be sufficient? Because otherwise I would wonder where a blank (but non-empty) PURL would come from... and consider it to be a bug in the origin of such a blank PURL.